File: 프로그래머스/0/120808. 분수의 덧셈/분수의 덧셈.py
def solution(numer1, denom1, numer2, denom2):
    answer = [0, 0]
    
    numer3 = denom2 * numer1 + numer2 * denom1
    denom3 = denom1 * denom2
    
    i = numer3
    
    for i in range(numer3, 0, -1):
        if numer3 % i == 0 and denom3 % i == 0:
            break
    
    answer[0] = numer3 / i
    answer[1] = denom3 / i

    return answer

File: 프로그래머스/0/120808. 분수의 덧셈/분수의 덧셈.java
class Solution {
    public int[] solution(int numer1, int denom1, int numer2, int denom2) {
        int[] answer = new int[2];
    
        int numer3 = denom2 * numer1 + numer2 * denom1;
        int denom3 = denom1 * denom2;

        int i = numer3;

        for (; i > 0; i--) {
            if (numer3 % i == 0 && denom3 % i == 0) {
                break;
            }
        }

        answer[0] = numer3 / i;
        answer[1] = denom3 / i;
        
        return answer;
    }
}